pulsar/producer_partition.go

```diff
@@ -197,30 +248,32 @@ func (p *partitionProducer) grabCnx() error {
 	}
 
 	p.producerName = res.Response.ProducerSuccess.GetProducerName()
+
+	var provider internal.BatcherBuilderProvider
 	if p.options.DisableBatching {
-		provider, _ := GetBatcherBuilderProvider(DefaultBatchBuilder)
-		p.batchBuilder, err = provider(p.options.BatchingMaxMessages, p.options.BatchingMaxSize,
-			p.producerName, p.producerID, pb.CompressionType(p.options.CompressionType),
-			compression.Level(p.options.CompressionLevel),
-			p,
-			p.log)
-		if err != nil {
-			return err
-		}
+		provider, _ = GetBatcherBuilderProvider(DefaultBatchBuilder)
+
 	} else if p.batchBuilder == nil {
-		provider, err := GetBatcherBuilderProvider(p.options.BatcherBuilderType)
```

Comment:
Why change this? The `p` masks the partitionProducer and make the code harder to follow.

Reply:
Just tried to remove duplicate code :) If it is not ok, I'll revert it.

Reply:
reverted